src/auth/oauth2client.ts

```diff
@@ -214,8 +214,12 @@ export class OAuth2Client extends AuthClient {
       grant_type: 'authorization_code'
     };
 
-    const res = await this.transporter.request<CredentialRequest>(
-        {method: 'POST', url, data: values});
+    const res = await this.transporter.request<CredentialRequest>({
+      method: 'POST',
+      url,
+      data: querystring.stringify(values),
+      headers: {'Content-Type': 'application/x-www-form-urlencoded'}
```

Comment:
Should there be a test modification that expects this header to be present?

Reply:
Yes :) I just included a change to the nock config in the test that will now make sure the content type is properly set.